tests/rules/import-alias.test.ts

```diff
@@ -3,6 +3,7 @@ import { rules } from "#src/index";
 import { RuleTester } from "eslint";
 import { existsSync } from "fs-extra";
 import { mocked } from "jest-mock";
+import { join as joinPath } from "path";
```

Comment:
The filenames in the rule test cases actually don't have to use Windows style separators, because the path module converts slashes to backslashes on Windows (`path.normalize`), so I have removed the `path.join` calls again.  The absolute paths in the alias config need to use native separators though.

I have tried to mock the `path` module and run the tests with Unix and Windows style separators, but I can't come up with a solution that works and passes ESLint.
This seems to work: https://gist.github.com/pablohoch/57aeabde2fbf2488a68ce92777447c79
However, ESLint doesn't accept the `const path = require("path");` line. Using `import` doesn't work, because then `path.sep` can't be changed. So far, I couldn't find another way to mock/override properties using jest that works.

Reply:
Thanks for looking into it. I posted onto your gist an alternative which seems to be okay with ESLint and TypeScript after working off your copy.

Namely, I use `import path= require("path")` to get around ESLint and `Object.assign(path, path[platform])` to get around TS (as TS was not happy with reassigning the readonly `sep`). There were some other refactoring changes that I added if you ran a diff against it though, but I don't care too much about them.

(I also would've totally been okay with disabling the ESLint rule for the require line though to be honest, especially if there weren't any alternatives.)

Reply:
Thanks. I've updated the PR with your version, but I have made two small changes, because otherwise some of the tests with Unix style paths fail on Windows:

`path.posix.resolve` and `path.posix.dirname` don't work with Windows path names (they remove the drive letter), but they are called with the result of `process.cwd()`. So instead of overriding all `path` methods with those from `path.posix`, I've changed it to only override `join` and `sep`.

Also, while `path.win32.join` works with slashes and backslashes, `path.posix.join` only accepts slashes. Therefore, I've added to code to replace backslashes with slashes before calling `path.posix.join`. I'm not sure if this is the best way to do it, but at least all the tests pass on Windows and Linux.

Reply:
Awesome, thanks! Merged.